### ISSUE TITLE
cdk8s init typescript-app compiles with errors

### ISSUE THREAD

**Issue by briancaffey** [cdk8s-team/cdk8s#580]
### Description of the bug:

Starting a new project results in compilation errors. 

### Reproduction Steps:

```
npm i -g cdk8s-cli
mkdir -p k8s/cdk8s && cd k8s/cdk8s
cdk8s init typescript-app
```

This results in the errors shown above. 

### Error Log:

```
cdk8s init typescript-app
Initializing a project from the typescript-app template

added 3 packages, and audited 11 packages in 1s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 671 packages, and audited 682 packages in 13s

65 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> cdk8s@1.0.0 import
> cdk8s import

k8s

> cdk8s@1.0.0 compile
> tsc

../../node_modules/@types/sizzle/index.d.ts:14:35 - error TS2304: Cannot find name 'Element'.

14     <TArrayLike extends ArrayLike<Element>>(selector: string, context: Element | Document | DocumentFragment, results: TArrayLike): TArrayLike;
                                     ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:14:72 - error TS2304: Cannot find name 'Element'.

14     <TArrayLike extends ArrayLike<Element>>(selector: string, context: Element | Document | DocumentFragment, results: TArrayLike): TArrayLike;
                                                                          ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:14:82 - error TS2304: Cannot find name 'Document'.

14     <TArrayLike extends ArrayLike<Element>>(selector: string, context: Element | Document | DocumentFragment, results: TArrayLike): TArrayLike;
                                                                                    ~~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:14:93 - error TS2304: Cannot find name 'DocumentFragment'.

14     <TArrayLike extends ArrayLike<Element>>(selector: string, context: Element | Document | DocumentFragment, results: TArrayLike): TArrayLike;
                                                                                               ~~~~~~~~~~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:15:34 - error TS2304: Cannot find name 'Element'.

15     (selector: string, context?: Element | Document | DocumentFragment): Element[];
                                    ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:15:44 - error TS2304: Cannot find name 'Document'.

15     (selector: string, context?: Element | Document | DocumentFragment): Element[];
                                              ~~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:15:55 - error TS2304: Cannot find name 'DocumentFragment'.

15     (selector: string, context?: Element | Document | DocumentFragment): Element[];
                                                         ~~~~~~~~~~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:15:74 - error TS2304: Cannot find name 'Element'.

15     (selector: string, context?: Element | Document | DocumentFragment): Element[];
                                                                            ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:18:28 - error TS2304: Cannot find name 'Element'.

18     matchSelector(element: Element, selector: string): boolean;
                              ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:19:41 - error TS2304: Cannot find name 'Element'.

19     matches(selector: string, elements: Element[]): Element[];
                                           ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:19:53 - error TS2304: Cannot find name 'Element'.

19     matches(selector: string, elements: Element[]): Element[];
                                                       ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:41:48 - error TS2304: Cannot find name 'Element'.

41             (match: RegExpMatchArray, context: Element | Document, isXML: boolean): Element[] | void;
                                                  ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:41:58 - error TS2304: Cannot find name 'Document'.

41             (match: RegExpMatchArray, context: Element | Document, isXML: boolean): Element[] | void;
                                                            ~~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:41:85 - error TS2304: Cannot find name 'Element'.

41             (match: RegExpMatchArray, context: Element | Document, isXML: boolean): Element[] | void;
                                                                                       ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:73:20 - error TS2304: Cannot find name 'Element'.

73             (elem: Element): boolean;
                      ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:81:24 - error TS2304: Cannot find name 'Element'.

81             (elements: Element[], argument: number, not: boolean): Element[];
                          ~~~~~~~

../../node_modules/@types/sizzle/index.d.ts:81:68 - error TS2304: Cannot find name 'Element'.

81             (elements: Element[], argument: number, not: boolean): Element[];
                                                                      ~~~~~~~


Found 17 errors.

Error: error during project initialization: Error: Command failed: npm run compile
    at checkExecSyncError (node:child_process:636:11)
    at execSync (node:child_process:672:15)
    at Object.exports.post (/home/brian/.nvm/versions/node/v15.3.0/lib/node_modules/cdk8s-cli/templates/typescript-app/.hooks.sscaff.js:25:3)
    at executePostHook (/home/brian/.nvm/versions/node/v15.3.0/lib/node_modules/cdk8s-cli/node_modules/sscaff/lib/sscaff.js:61:37)
    at Object.sscaff (/home/brian/.nvm/versions/node/v15.3.0/lib/node_modules/cdk8s-cli/node_modules/sscaff/lib/sscaff.js:28:15)
    at async Object.handler (/home/brian/.nvm/versions/node/v15.3.0/lib/node_modules/cdk8s-cli/lib/cli/cmds/init.js:34:13)
STDOUT:
undefined
STDERR:
undefined
    at Object.handler (/home/brian/.nvm/versions/node/v15.3.0/lib/node_modules/cdk8s-cli/lib/cli/cmds/init.js:37:19)
```

### Environment:

  - **Framework Version**: 

```
npm list -g cdk8s-cli
/home/brian/.nvm/versions/node/v15.3.0/lib
└── cdk8s-cli@1.0.0-beta.11
```
  - **OS**: WSL2 Ubuntu 20.04

### Other:

This solution proposed in this GH issue comment fixes the issue: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/46167#issuecomment-660489799

In `tsconfig.json` I added `dom` to the `lib` array. Should this be added, or is there possibly something else that I should change?

--- 

This is :bug: Bug Report

**Comment by krubenrc** [comment]
Hello,
I too have errors when start a new project.

With this docker-compose:

`version: '3'
services:
    nodejs:
        image: node:14-buster
        stdin_open: true
        tty: true
        volumes:
            - ./app:/app
        working_dir: /app`

I installed the cdk8s-cli like the docs say: npm install -g cdk8s-cli
And i create a new project: cdk8s init typescript-app

And i get the next errors:

![image](https://user-images.githubusercontent.com/33639842/120381465-be10ab80-c322-11eb-87cb-a22e25e666bf.png)


Regards.

**Comment by eladb** [comment]
This has already been resolved by https://github.com/cdk8s-team/cdk8s-cli/pull/8. We are planning to release a fix shortly and will report back on this issue.

**Comment by eladb** [comment]
Released cdk8s-cli@1.0.0-beta.19
